### PR TITLE
o/servicestate/quota_control_test.go: change helper escaping

### PR DESCRIPTION
**Pull request by anonymouse64**
This puts the responsibility of escaping into the helpers and makes it a bit
easier to write tests for group names that need to be escaped.

Also modify one of the test cases to make sure that we have coverage of group
names with escaped characters in the name.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thanks!

File: overlord/servicestate/quota_control_test.go [hunk 1]
```diff
@@ -127,6 +128,7 @@ func checkQuotaState(c *C, st *state.State, exp map[string]quotaGroupState) {
 }
 
 func checkSvcAndSliceState(c *C, snapSvc string, sliceName string, sliceMem quantity.Size) {
```
Comment (pedronis) [inline overlord/servicestate/quota_control_test.go:130]: should the input be called slicePath then?

File: overlord/servicestate/quota_control_test.go [hunk 2]
```diff
@@ -113,7 +114,7 @@ func checkQuotaState(c *C, st *state.State, exp map[string]quotaGroupState) {
 				// meh assume all services are named svc1
 				sliceName := name
```
Comment (pedronis) [inline overlord/servicestate/quota_control_test.go:115]: same for this var

**Review by pedronis — APPROVED** [review]
thank you, nitpick about naming

File: overlord/servicestate/quota_control_test.go (hunk 2 above)
Comment (anonymouse64) [reply overlord/servicestate/quota_control_test.go]: done

File: overlord/servicestate/quota_control_test.go (hunk 1 above)
Comment (anonymouse64) [reply overlord/servicestate/quota_control_test.go]: done

**Pull request merged**